src/main/java/appeng/items/parts/ItemMultipartRendering.java

```diff
@@ -0,0 +1,33 @@
+package appeng.items.parts;
+
+
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.item.ItemStack;
+
+import appeng.bootstrap.IItemRendering;
+import appeng.bootstrap.ItemRenderingCustomizer;
+
+
+public class ItemMultipartRendering extends ItemRenderingCustomizer
+{
+
+	private final ItemMultiPart item;
+
+	public ItemMultipartRendering( ItemMultiPart item )
+	{
+		this.item = item;
+	}
+
+	@Override
+	public void customize( IItemRendering rendering )
+	{
+		rendering.meshDefinition( this::getItemMeshDefinition );
+	}
+
+	private ModelResourceLocation getItemMeshDefinition( ItemStack is )
+	{
+		// TODO: Avoid object creation here. This happens every frame
```

Comment:
Questionable, the JVM can create an object on the stack and not heap when applicable. Thus without real benchmarking it is hard to predict.